File: logisim/app/src/main/java/logisim/gui/log/Loggable.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.log;

import logisim.circuit.CircuitState;
import logisim.data.WireValue.WireValue;

public interface Loggable {
	Object[] getLogOptions(CircuitState state);

	String getLogName(Object option);

	WireValue getLogValue(CircuitState state, Object option);
}

File: LogisimPlus/LogisimPlus/logisim/gui/log/Loggable.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.log
{
	using CircuitState = logisim.circuit.CircuitState;
	using Value = logisim.data.Value;

	public interface Loggable
	{
		object[] getLogOptions(CircuitState state);

		string getLogName(object option);

		Value getLogValue(CircuitState state, object option);
	}

}
